### PR TITLE
refactor: dwh proto types

### PR DESCRIPTION
**Pull request by oopcode**

### CODE REVIEW

File: insonmnia/dwh/server.go [hunk 1]
```diff
@@ -84,7 +84,7 @@ func NewDWH(ctx context.Context, cfg *Config, key *ecdsa.PrivateKey) (*DWH, erro
 
 func (w *DWH) Serve() error {
 	if w.cfg.Blockchain != nil {
-		bch, err := blockchain.NewAPI(blockchain.WithEthEndpoint(w.cfg.Blockchain.EthEndpoint))
+		bch, err := blockchain.NewAPI(blockchain.WithSidechainEndpoint(w.cfg.Blockchain.EthEndpoint))
```
Comment (antmat) [inline insonmnia/dwh/server.go]: Is this ok?

File: insonmnia/dwh/server.go [hunk 2]
```diff
@@ -157,19 +157,19 @@ func (w *DWH) getDeals(ctx context.Context, request *pb.DealsRequest) (*pb.DWHDe
 	if request.Status > 0 {
 		filters = append(filters, newFilter("Status", eq, request.Status, "AND"))
 	}
-	if request.SupplierID > "0" {
-		filters = append(filters, newFilter("SupplierID", eq, request.SupplierID, "AND"))
+	if request.SupplierID != nil && request.SupplierID.Unwrap().Hex() > "0x0" {
```
Comment (antmat) [inline insonmnia/dwh/server.go]: Introduce IsZero for EthAddr

File: insonmnia/dwh/server.go [hunk 3]
```diff
@@ -157,19 +157,19 @@ func (w *DWH) getDeals(ctx context.Context, request *pb.DealsRequest) (*pb.DWHDe
...
+	if request.ConsumerID != nil && request.ConsumerID.Unwrap().Hex() > "0x0" {
+		filters = append(filters, newFilter("ConsumerID", eq, request.ConsumerID.Unwrap().Hex(), "AND"))
 	}
-	if request.MasterID > "0" {
-		filters = append(filters, newFilter("MasterID", eq, request.MasterID, "AND"))
+	if request.MasterID != nil && request.MasterID.Unwrap().Hex() > "0x0" {
+		filters = append(filters, newFilter("MasterID", eq, request.MasterID.Unwrap().Hex(), "AND"))
 	}
-	if request.AskID > "0" {
+	if request.AskID != nil && request.AskID.Unwrap().String() > "0" {
```
Comment (antmat) [inline insonmnia/dwh/server.go]: use IsZero()

File: insonmnia/dwh/server.go [hunk 4]
```diff
@@ -157,19 +157,19 @@ func (w *DWH) getDeals(ctx context.Context, request *pb.DealsRequest) (*pb.DWHDe
...
 		filters = append(filters, newFilter("AskID", eq, request.AskID, "AND"))
 	}
-	if request.BidID > "0" {
+	if request.BidID != nil && request.BidID.Unwrap().String() > "0" {
```
Comment (antmat) [inline insonmnia/dwh/server.go]: here too

File: insonmnia/dwh/server.go (hunk 1 above)
Comment (oopcode) [reply insonmnia/dwh/server.go]: yes, it's the fixed version

File: insonmnia/dwh/server.go [hunk 5]
```diff
@@ -1413,7 +1420,7 @@ func (w *DWH) onOrderUpdated(orderID *big.Int) error {
 	}
 
 	// If order was updated, but no deal is associated with it, delete the order.
-	if order.DealID <= "0" {
+	if order.DealID != nil && order.DealID.Unwrap().String() <= "0" {
```
Comment (antmat) [inline insonmnia/dwh/server.go]: here

File: insonmnia/structs/task.go [hunk 6]
```diff
@@ -22,7 +22,7 @@ func NewStartTaskRequest(request *sonm.StartTaskRequest) (*StartTaskRequest, err
 		return nil, errDealRequired
 	}
 
-	if deal.GetId() == "" {
+	if deal.GetId().Unwrap().String() < "0" {
```
Comment (antmat) [inline insonmnia/structs/task.go]: here

File: proto/insonmnia.go [hunk 7]
```diff
@@ -72,6 +72,14 @@ func (m *EthAddress) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	return nil
 }
 
+func (m *EthAddress) IsZero() bool {
+	return NewBigInt(m.Unwrap().Big()).IsZero()
```
Comment (antmat) [inline proto/insonmnia.go]: no need in BigInt just
m.Unwrap().Big().Bitlen() == 0

Comment (antmat) [reply proto/insonmnia.go]: also nil receiver compat would be great

**Pull request merged**